src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/NotifyRazorEditorCreated.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    /// <summary>
+    /// Used for internal Visual Studio infrastructure. Not for public consumption.
+    /// </summary>
+    public abstract class NotifyRazorEditorCreated
+    {
+        /// <summary>
+        /// Used for internal Visual Studio infrastructure. Not for public consumption.
+        /// </summary>
+        public abstract Task AddToWorkspaceAsync(IVsHierarchy hierarchy, uint itemId);
```

Comment:
@ToddGrun @alexgav does this seem like what you'd expect?

Reply:
@rynowak, any reason this couldn't live in the Microsoft.VisualStudio.Editor.Razor.dll? I imagine we'd want to share the logic between vswindows and vsmac

Reply:
because `IVsHierarchy` 😆 

For now we're going to put these function behind a feature flag in windows VS. This is the first step of a long set of things we're going to iterate one.